[UpSample nearest]

jjsjann123 opened the pull request:
1. update on restricting block.z <= 64, compliant to CUDA maximum z-dimension of
a block;
2. clang-format

jjsjann123 committed:
[UpSample nearest]

1. update on restricting block.z <= 64, compliant to CUDA maximum z-dimension of
a block;
2. clang-format

ngimel commented:
test case please?

ngimel commented on aten/src/ATen/native/cuda/UpSampleNearest2d.cu:
```diff
@@ -147,7 +151,8 @@ static void upsample_nearest2d_out_cuda_template(
   int block_y = std::min<int>(lastPow2(output_height), max_threads / block_x);
-  int block_z = std::min<int>(nc, max_threads / block_x / block_y);
+  int block_z =
+      std::min<int>(64, std::min<int>(nc, max_threads / block_x / block_y));
```
Can you please query maxThreadDims?  
https://docs.nvidia.com/cuda/cuda-runtime-api/structcudaDeviceProp.html#structcudaDeviceProp_192d195493a9d36b2d827aaf3ffd89f1e

soumith reviewed (changes requested):
waiting for a test case

jjsjann123 committed:
address review comments:

1. Switch to maxThreadsDim to restrict CTA config;
2. added test to cover previous failing config for upsample nearest;

jjsjann123 commented:
Test case added to test_cuda.py.
Verified test failure without the fix.

jjsjann123 commented:
FYI, We should close #22526 after merging this one.

jjsjann123 committed:
moving test from test_cuda.py to test_nn

ngimel reviewed (approved):
There's still lint.

jjsjann123 commented:
Lint complains about something in the generated file: 
```$ flake8
./test/test_quantization.py:4:1: F401 'torch.quantization as tq' imported but unused
The command "flake8" exited with 1.```

Maybe it's caused by the added test in common_nn.py? I don't see anything wrong there tho and lint doesn't give enough hint :/

ngimel commented:
Yeah, it's probably unrelated then.

ngimel commented:
@pytorchbot rebase this please

Commit:
Merge remote-tracking branch 'origin/master' into HEAD

jjsjann123 commented:
OK, flake8 seems to be very specific about one line that is >120, while it's OK with other lines in the same manner in that function. There seems to be something funky there.

I can change that line just to bypass the pylinter. But I don't think it's relevant to this PR and probably would conflict with other PRs doing the same thing. Maybe we should just ignore it here?

ngimel commented:
linter is broken by recently merged #21860, it has a few lines that are longer than 120 characters.

ngimel commented:
@pytorchbot merge this please

ngimel commented:
lint should now be fixed, all other tests are green.
